cmd/proxy/actions/cache_miss_fetcher.go

```diff
@@ -61,7 +61,21 @@ func parseArgs(args worker.Args) (string, string, error) {
 
 func getModuleInfo(ctx context.Context, module, version string) (*storage.Version, error) {
 	os := olympusStore.NewStorage(GetOlympusEndpoint())
-	return os.Get(ctx, module, version)
+	var v storage.Version
+	var err error
+	v.Info, err = os.Info(ctx, module, version)
+	if err != nil {
```

Comment:
can we do this async, and with multierror?

Reply:
also we need to start thinking about concurrent cache fills,

Reply:
@michalpristas another clean up question: do we still want the cache_miss_fetcher? or should it be deleted?

Reply:
i think we will need it in a modified form.
From the flow described in #346 we talked that proxy on a miss redirects to Olympus and schedules a job to populate itself from O after some (configured) time. 
This might be a place where cache_miss_fetcher might be of use

Reply:
@michalpristas is it ok to do that in a separate PR? This pr is already big and bulky. I'll push one right after :)

Reply:
yes, i will always 👍 for smaller, more readable PRs